Source/Core/DolphinQt2/HotkeyScheduler.cpp

```diff
@@ -0,0 +1,353 @@
+      {
+        if (g_Config.iStereoMode != STEREO_ANAGLYPH)
+        {
+          g_Config.iStereoMode = STEREO_ANAGLYPH;
+          g_Config.sPostProcessingShader = DUBOIS_ALGORITHM_SHADER;
+        }
+        else
+        {
+          g_Config.iStereoMode = STEREO_OFF;
+          g_Config.sPostProcessingShader = "";
+        }
+      }
+
+      if (IsHotkey(HK_TOGGLE_STEREO_3DVISION))
+      {
+        if (g_Config.iStereoMode != STEREO_3DVISION)
+        {
+          if (g_Config.sPostProcessingShader == DUBOIS_ALGORITHM_SHADER)
+            g_Config.sPostProcessingShader = "";
+
+          g_Config.iStereoMode = STEREO_3DVISION;
+        }
+        else
+        {
+          g_Config.iStereoMode = STEREO_OFF;
+        }
+      }
+    }
+
+    if (IsHotkey(HK_DECREASE_DEPTH, true))
+      g_Config.iStereoDepth = std::max(g_Config.iStereoDepth - 1, 0);
+
+    if (IsHotkey(HK_INCREASE_DEPTH, true))
+      g_Config.iStereoDepth = std::min(g_Config.iStereoDepth + 1, 100);
+
+    if (IsHotkey(HK_DECREASE_CONVERGENCE, true))
+      g_Config.iStereoConvergence = std::max(g_Config.iStereoConvergence - 5, 0);
+
+    if (IsHotkey(HK_INCREASE_CONVERGENCE, true))
+      g_Config.iStereoConvergence = std::min(g_Config.iStereoConvergence + 5, 500);
+
+    // Freelook
+    static float fl_speed = 1.0;
+
+    if (IsHotkey(HK_FREELOOK_DECREASE_SPEED, true))
+      fl_speed /= 1.1f;
+
+    if (IsHotkey(HK_FREELOOK_INCREASE_SPEED, true))
+      fl_speed *= 1.1f;
+
+    if (IsHotkey(HK_FREELOOK_RESET_SPEED, true))
+      fl_speed = 1.0;
+
+    if (IsHotkey(HK_FREELOOK_UP, true))
+      VertexShaderManager::TranslateView(0.0, 0.0, -fl_speed);
+
+    if (IsHotkey(HK_FREELOOK_DOWN, true))
+      VertexShaderManager::TranslateView(0.0, 0.0, fl_speed);
+
+    if (IsHotkey(HK_FREELOOK_LEFT, true))
+      VertexShaderManager::TranslateView(fl_speed, 0.0);
+
+    if (IsHotkey(HK_FREELOOK_RIGHT, true))
+      VertexShaderManager::TranslateView(-fl_speed, 0.0);
+
+    if (IsHotkey(HK_FREELOOK_ZOOM_IN, true))
+      VertexShaderManager::TranslateView(0.0, fl_speed);
+
+    if (IsHotkey(HK_FREELOOK_ZOOM_OUT, true))
+      VertexShaderManager::TranslateView(0.0, -fl_speed);
+
+    if (IsHotkey(HK_FREELOOK_RESET, true))
+      VertexShaderManager::ResetView();
+
+    // Savestates
+    for (u32 i = 0; i < State::NUM_STATES; i++)
+    {
+      if (IsHotkey(HK_LOAD_STATE_SLOT_1 + i))
+        State::Load(i + 1);
+
+      if (IsHotkey(HK_SAVE_STATE_SLOT_1 + i))
+        State::Save(i + 1);
+
+      if (IsHotkey(HK_LOAD_LAST_STATE_1 + i))
+        State::LoadLastSaved(i + 1);
+
+      if (IsHotkey(HK_SELECT_STATE_SLOT_1 + i))
+        emit SetStateSlotHotkey(i + 1);
+    }
+
+    if (IsHotkey(HK_SAVE_FIRST_STATE))
+      State::SaveFirstSaved();
+
+    if (IsHotkey(HK_UNDO_LOAD_STATE))
+      State::UndoLoadState();
+
+    if (IsHotkey(HK_UNDO_SAVE_STATE))
+      State::UndoSaveState();
+  }
+
+  Common::SleepCurrentThread(1000 / 60);
```

Comment:
I think this is outside of the "while stop-flag isn't set" loop?

Also this will cause a busy loop when the HotkeyManager isn't enabled. Probably easiest to move this to the very top of the loop.